File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] suf = new int[nums.length];
        int[] pre = new int[nums.length];
        suf[0] = nums[0];
        pre[nums.length-1] = nums[nums.length-1];
        for(int i = 1;i<nums.length;i++){
            suf[i]=suf[i-1]*nums[i];
            pre[nums.length-1-i]=pre[nums.length-i]*nums[nums.length-1-i];
        }
        nums[0]=pre[1];
        nums[nums.length-1]=suf[nums.length-2];
        for(int i = 1;i<nums.length-1;i++){
            nums[i]=suf[i-1]*pre[i+1];
        }
        return nums;
    }
}  

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent( vector<int>& nums, int k) {
        unordered_map<int, int> mp{};
        for(const auto num : nums){
            mp[num]++;
        }
        vector<vector<int>> arr(nums.size()+1);
        for(const auto & [key, value] : mp){
            arr[value].push_back(key);
        }
        vector<int> result;
        for(auto i{arr.size()};i-->0;){
            if(!arr[i].empty()){
                for(const auto num : arr[i]){
                    if(k-->0){
                        result.push_back(num);
                    } else return result;
                }
            }
        }
        return result;
    }
};
